pkg/api/errors/placement.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieh.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+import (
+	"fmt"
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+
+	kiterrors "github.com/dapr/kit/errors"
+)
+
+const (
+	PlacementServiceClosed         = "CLOSED"
+	PlacementServiceAlreadyRunning = "ALREADY_RUNNING"
+
+	CodePrefixPlacement = "DAPR_PLACEMENT_"
+)
+
+func PlacementServiceIsClosedOnRun(msg string) error {
+	return kiterrors.NewBuilder(
+		codes.Unavailable,
+		http.StatusExpectationFailed,
+		fmt.Sprintln(msg),
+		CodePrefixPlacement+kiterrors.CodePostfixQueryFailed,
+	).WithErrorInfo(CodePrefixPlacement+kiterrors.CodePostfixQueryFailed, nil).
+		Build()
+}
+
+func PlacementServiceIsAlreadyRunning(msg string) error {
+	return kiterrors.NewBuilder(
+		codes.AlreadyExists,
+		http.StatusExpectationFailed,
+		fmt.Sprintln(msg),
```

Comment:
```suggestion
		msg,
```

Reply:
Thanks for the review and tips @elena-kolevska: regarding the integration tests, i've been trying to figure out what to do from mtls.go and follower.go, but I'm not able to figure how to force the errors for the Service struct in the integration tests. I'm probably just missing some understanding, but I'm keen to get it right, if I can get some tips to bridge my understanding.

Reply:
I am still keen to fix these integration tests, but am stuck with how to force the errors, as stated previously. If I could get some advice I would jump on it asap.

Reply:
Hey @fowlerlee , did you see my previous comment where I linked to an example I prepared for the follower error?
https://github.com/elena-kolevska/dapr/commit/81a00b3c1eb53f59ccbdbc59050080070ffa383d
Testing some of these errors is indeed quite complex, I can take a look at the PR again and see if I can find a way to force that Sentry error, for example.

Reply:
@fowlerlee can we address the outstanding comments about the code first before proceeding to the tests? Otherwise, we might end up having to change the tests again after we change the code.
For example, the one about the different errors in `validateClient` is really important.

Reply:
@elena-kolevska, I have pushed fixes for outstanding comments now. It's on my branch and ready for review. Thank you for your time.